lib/webhdfs/client_v1.rb

```diff
@@ -384,6 +430,19 @@ def request(host, port, method, path, op=nil, params={}, payload=nil, header=nil
                     'Response body is empty...'
                   end
 
+        # when delegation token is expired
+        if res.code == '403' and @renew_kerberos_delegation_token_time_hour
+          if message.include?('{"RemoteException":{')
+            detail = JSON.parse(message) rescue nil
+            if detail.dig('RemoteException', 'exception') == 'InvalidToken' and detail.dig('RemoteException', 'message').include?('HDFS_DELEGATION_TOKEN')
+              @kerberos_delegation_token = get_kerberos_delegation_token(@username)
```

Comment:
Shouldn't we update `@kerberos_token_updated_at` here?
Or, it may be better to have an option on `get_cached_kerberos_delegation_token` to get and update the token forcibly.

Reply:
I made an option for force. thank you for your kind feedback!! It's really good to me :)